CONTRIBUTING.md

```diff
@@ -0,0 +1,121 @@
+# Contributing
+
+Welcome! We're excited that you're interested in contributing. Below are some
+basic guidelines.
+
+## Workflow
+
+Dskit follows a standard GitHub pull request workflow. If you're unfamiliar
+with this workflow, read the very helpful [Understanding the GitHub
+flow][github-flow] guide from GitHub.
+
+[github-flow]: https://guides.github.com/introduction/flow/
+
+You are welcome to create draft PRs at any stage of readiness - this can be
+helpful to ask for assistance or to develop an idea. But before a piece of work
+is finished it should:
+
+* Be organised into one or more commits, each of which has a commit message
+  that describes all changes made in that commit ('why' more than 'what' - we
+  can read the diffs to see the code that changed).
+
+* Each commit should build towards the whole - don't leave in back-tracks and
+  mistakes that you later corrected.
+
+* Have unit tests for new functionality or tests that would have caught the bug
+  being fixed.
+
+* Include a CHANGELOG message if users of Dskit need to hear about what you
+  did.
+
+## Testing und Linting the Code
+
+To lint the code:
+
+```bash
+make lint
+```
+
+To run the unit tests suite:
+
+```bash
+make test
+```
+
+### Dependency management
+
+We use [go modules] to manage dependencies on external packages.  This requires
+a working Go environment with version 1.16 or greater, git and [bzr] installed.
+
+[Go modules]: https://golang.org/cmd/go/#hdr-Modules__module_versions__and_more
+[bzr]: http://wiki.bazaar.canonical.com/Download
+
+To add or update a new dependency, use the `go get` command:
+
+```bash
+# Pick the latest tagged release.
+go get example.com/some/module/pkg
+
+# Pick a specific version.
+go get example.com/some/module/pkg@vX.Y.Z
+```
+
+Tidy up the `go.mod` and `go.sum` files:
+
+```bash
+go mod tidy
```

Comment:
should we include `go mod vendor` here too? Otherwise lint check will fail.

Reply:
We currently don't have a vendor folder checked in and hence we also don't do a lint check on it.

https://github.com/grafana/dskit/blob/4710207529678a6e5f8954d1c280b1d099a3a70e/Makefile#L27-L32

Reply:
Christian is right I think.

Reply:
I see. I thought we follow the same practice as in Cortex, but forgot to check. Thanks!